### PR TITLE
add object detection canary

### PR DESCRIPTION
**Pull request by marthajohnston**
This change adds a canary test for object detection to run in parallel with the existing canary test for video stream. In addition to the code changes, I also added a transform camera in the robot being used in app.viam. The two tests will be shown with a split screen on the cv desktop monitor. Below is a video showing the current behavior.

https://user-images.githubusercontent.com/61131470/219798852-daa58fbb-1916-47d9-916e-abf89e747b5e.mov

### CODE REVIEW

**Comment by bazile-clyde** [comment]
It works!! The only open concern I have is how much it slows things down. If the overall goal is just to check our detection models work we could call `get_detections` like we do in the [rgb_follower.py](https://github.com/viam-labs/line-follower/blob/main/rgb_follower.py) script. We'd then only have one call to `get_images`, which I'm assuming is the bottleneck. `get_detections` would return the xy coordinates of the rectangle and a label and we could overlay that on one stream and not the other. @bhaney thoughts? How important do you think it is to test the pipeline behavior as well?

**Comment by marthajohnston** [comment]
@bhaney can you also log in to the cv desktop pi and check the crontab?

**Comment by bhaney** [comment]
> It works!! The only open concern I have is how much it slows things down. If the overall goal is just to check our detection models work we could call `get_detections` like we do in the [rgb_follower.py](https://github.com/viam-labs/line-follower/blob/main/rgb_follower.py) script. We'd then only have one call to `get_images`, which I'm assuming is the bottleneck. `get_detections` would return the xy coordinates of the rectangle and a label and we could overlay that on one stream and not the other. @bhaney thoughts? How important do you think it is to test the pipeline behavior as well?

So I think we do want to actually benchmark the detector transform camera with the canary, rather than overlaying our the detection bounding boxes ourselves. Transform cameras are interesting because they depend on two things: both the source camera, and the vision service. If this dependency structure breaks, it would be really good to see. 

And yea, the slowdown is a bit concerning, but wouldn't this be a standard use case? Someone streams their images fro ma webcam, and applies an ML model to overlay something on it as well. I think this shows a very real world application about how two cameras running on two threads consumes the CPU and slows things down.

**Comment by bazile-clyde** [comment]
> I think this shows a very real world application about how two cameras running on two threads consumes the CPU and slows things down.

@bhaney Agreed. But then we wouldn't know how many fps we could get for a single 2D stream. We know we can stream at about 13-14 fps because we see it every day. If this change were pushed as is we'd lose that info. That's my only concern.

**Review by bhaney — APPROVED** [review]
This looks good to me!

**Comment by bhaney** [comment]
> > I think this shows a very real world application about how two cameras running on two threads consumes the CPU and slows things down.
> 
> @bhaney Agreed. But then we wouldn't know how many fps we could get for a single 2D stream. We know we can stream at about 13-14 fps because we see it every day. If this change were pushed as is we'd lose that info. That's my only concern.

Then maybe we alternate streams? One just webcam, then just just detector, and then both? Which cycles over 3 hours.

**Review by bhaney — CHANGES_REQUESTED** [review]
Actually, maybe as an alternative, you create an option for the update script: Based on an argument, 2D_image_stream.py will either display:

1. Just the 2d webcam stream
2. just the detector stream
3. Both side by side

**Comment by bazile-clyde** [comment]
> 3\. Both side by side

That could work. I wonder if we could just add another webcam to the Pi too. So you can still get a split screen but from different cameras. That wouldn't be as slow as this version but I'm not sure if it'd be just as fast as the single stream from one camera. I _think_ the Pi is fast enough to handle two cameras, no problem, but it's just a hunch. I'd be fine with any of the solutions so far.

**Comment by marthajohnston** [comment]
I've added the 3 different options, which is set through the crontab. I found that the major slowdown in the fps was due to the virtual pi desktop I was using on my mac, and once running it on the actually desktop pi it sped back up to normal speeds when only the regular stream was running. When both are running side by side, the normal stream does slow down to ~11-13 fps, so not as extreme as before but still not perfect. Let me know if you have any more thoughts

File: canaries/2D_image_stream.py [hunk 1]
```diff
@@ -9,12 +9,25 @@
...
 from viam.robot.client import RobotClient
 from viam.rpc.dial import Credentials, DialOptions
+from viam.components.camera.client import CameraMimeType
 
 import logging
+import requests
+import os
 
+url = os.getenv('MY_SLACK_WEBHOOK')
+arg = int(os.getenv('ARG'))
```
Comment (bazile-clyde) [inline canaries/2D_image_stream.py]: [opt] Alternatively you could use [command line args](https://www.tutorialspoint.com/python/python_command_line_arguments.htm) but this is fine.

File: canaries/2D_image_stream.py [hunk 2]
```diff
@@ -39,44 +52,101 @@ def get_frames_per_sec(ordered_frames):
...
+            await image_stream(robot, llist1)
+        elif arg == 2:
+            await detection_stream(robot, llist2)
+        elif arg == 3:
+            task1 = asyncio.create_task(image_stream(robot, llist1))
+            task2 = asyncio.create_task(detection_stream(robot, llist2))
+            await task1
+            await task2
+        else:
+            print("unaccepted argument")
```
Comment (bazile-clyde) [inline canaries/2D_image_stream.py]: [minor] Cool! The only thing I'm wondering is, do the screen sizes change based on the arguments too? It looks like `image_stream` and `detection_stream` are hard-coded to take up half the screen, which works well if they're concurrently displayed but not as well if only one is active.

**Review by bazile-clyde — CHANGES_REQUESTED** [review]

File: canaries/2D_image_stream.py (hunk 2 above)
Comment (marthajohnston) [reply canaries/2D_image_stream.py]: if one is active it is fullscreen

**Comment by bazile-clyde** [comment]
> I've added the 3 different options, which is set through the crontab. I found that the major slowdown in the fps was due to the virtual pi desktop I was using on my mac, and once running it on the actually desktop pi it sped back up to normal speeds when only the regular stream was running. When both are running side by side, the normal stream does slow down to ~11-13 fps, so not as extreme as before but still not perfect. Let me know if you have any more thoughts

That's great! I only have one [minor] comment and one [opt] comment. Everything else looks good to me.

I'm trying to follow the tags in https://viam.atlassian.net/wiki/spaces/ENG/pages/262371/Viam+Engineering+Guidelines#Pull-Requests.

File: canaries/2D_image_stream.py (hunk 2 above)
Comment (bazile-clyde) [reply canaries/2D_image_stream.py]: Ah, I see!

**Review by bazile-clyde — APPROVED** [review]

**Review by bhaney — APPROVED** [review]
Awesome! LGTM

**Pull request merged**